lib/helpers/getOneBlockPerCondition.m

```diff
@@ -0,0 +1,33 @@
+% Want a function which, given an input argument of the block number,
+% returns the condition of that block (medical or monetary) and the block's
+% beneficiary (self or other).
+
+
+function [ uniqueIdx ] = getOneBlockPerCondition(blocks)
+
+% adding paths?
+addpath(genpath('./lib'));
+addpath(genpath('./tasks/SODM'));
+
+% blocks and numBlocks
+numBlocks = length(blocks);
+
+payoffKinds = cell.empty;
+beneficiaries = cell.empty;
+
+% iterate through the blocks
+for blockIdx = 1:numBlocks
+    subsetofblocs = blocks{blockIdx};
+    conditions = subsetofblocs.conditions;
+    
+    payoffKinds{blockIdx} = conditions.payoffKind;
+    beneficiaries{blockIdx} = conditions.beneficiary;
```

Comment:
The way to generalize that for tasks with different kinds of conditions would be to a) poll fieldnames in `block.conditions` and make them into table headers, b) populate the table with the condition assignments extracted from each block. This is perfect for a vectorized approach -- in fact, `cellfun(@(block) struct2table(block.conditions), blocks)` would be a one-line expression of that logic (haven't checked).